source/Plugins/Language/Swift/SwiftHashedContainer.cpp

```diff
@@ -493,8 +493,6 @@ NativeHashedStorageHandler::NativeHashedStorageHandler(
   if (!scale_sp)
     return;
   auto scale = scale_sp->GetValueAsUnsigned(0);
-  if (scale > m_ptr_size)
-    return;
```

Comment:
Hm, I don't think we found out why this was failing. Scales larger than 31/63 would wreak havoc in `GetBucketCount`:

```c++
  // Line 88
  size_t GetBucketCount() { return 1 << m_scale; }
```

(Related: this should probably be changed to read `1ULL << m_scale`, as well.)

Reply:
This is `m_ptr_size` in /bytes/, i.e. 4. Should we have a check for this (but in bits?)

Reply:
i.e. `(scale > m_ptr_size * 8)`

Reply:
Argh, of course. 🤭 That's simultaneously a relief and a huge face palm.

This wasn't a super important condition -- it was just a sanity check against trying to interpret corrupt data. (In a valid Set/Dictionary, the scale should be between 0 and 32/64, depending on pointer size in bits.) I think we can safely omit it; subsequent code will fail when it tries to access memory anyway.

Reply:
Great :)